### DISCUSSION TITLE
Is there a configuration option to set LC_CTYPE?

### DISCUSSION THREAD

**Discussion by anttikissa** [kovidgoyal/kitty#3345] [General]
First of all, I'm on macOS 11.1 and kitty version 0.19.3.

I just changed to kitty and noticed that when running kitty from the system launcher (as opposed from another terminal, where LC_CTYPE is already set in environment), LC_CTYPE is not set to UTF-8, and this caused some things to break. In particular, vim thinks that my .vimrc is encoded in latin1 (which it isn't), leading to errors.

I can work around this by setting LC_CTYPE=UTF-8 in .zshrc, but that only works for my shell - for instance when using the Preferences menu option to edit kitty config file, I get the same .vimrc error (I took me a while to realize that this happens because encoding issues):

![image](https://user-images.githubusercontent.com/141618/109153161-317bad80-7775-11eb-8000-eaece8bfe0fd.png)

I looked into how other terminals set LC_CTYPE, and found settings for it in Terminal.app and iTerm2:

Terminal has Profiles -> Advanced -> Set locale environment variables on startup:
![image](https://user-images.githubusercontent.com/141618/109153285-625be280-7775-11eb-917c-820fe14b3eab.png)

iTerm2 has Profiles -> Terminal -> Set locale variables automatically
![image](https://user-images.githubusercontent.com/141618/109153689-e615cf00-7775-11eb-8e94-2cb123466762.png)

Unchecking these options unsets LC_CTYPE, and running `locale` shows `LC_CTYPE="C"`. And I get the same vim error.

As far as I can remember, I have never touched these options, so I assume UTF-8 is the default nowadays. My memory could fail me though.

Is there a configuration option to set LC_CTYPE to a given value? It might even make sense to default this to UTF-8, if the system doesn't provide a default value for this. (Not sure about different systems and languages, but as an European Mac user that would seem like a reasonable default.)

**Comment by kovidgoyal** [comment]
Use the env directive in kitty.conf to set any env var you like.

**Reply by anttikissa** [reply]
Ah, that did it - I had somehow overlooked the env directive. Thank you!

I got so far as to install a plist file according to [this StackOverflow answer](https://apple.stackexchange.com/a/413773/3253) but this is a much simpler way.
